py2many/rewriters.py

```diff
@@ -262,3 +263,43 @@ def visit_Call(self, node):
                     else:
                         return self._do_other_rewrite(node)
         return node
+
+
+class StrStrRewriter(ast.NodeTransformer):
+    def __init__(self, language):
+        super().__init__()
+        self._language = language
+
+    def visit_Compare(self, node):
+        if self._language in {"dart", "kotlin", "nim"}:
+            return node
+
+        if isinstance(node.ops[0], ast.In):
+            left = node.left
+            right = node.comparators[0]
+            left_type = get_id(get_inferred_type(left))
+            right_type = get_id(get_inferred_type(right))
+            if left_type == "str" and right_type == "str":
```

Comment:
how is this any better than the same logic appearing in each transpiler?  e.g. transpiling as `a.contains(b)` in Rust is desirable to most use of `a in b`, not just `str` & `str`.  Me thinks this is taking the "rewriter" concept to far.

Reply:
That said, there is a need for a generic rewriter, to convert `a.find(b) != -1` into `b in a` ;-)

Reply:
Yeah - 4-5 per language rewriters would be cleaner. But this was a lot fewer lines of code, which is not always the right metric to optimize for.

I'm for getting this in first and breaking it up into N rewriters if this pattern becomes hard to maintain and sets a bad example.

Reply:
I dont get why it needs any rewriters.  I would put it in `visit_Compare`.

> if this pattern ... sets a bad example.

IMO using `_language` in this way does set a bad example.  Using `_language` for when one language is very odd seems reasonable, but when a rewriter needs special logic for various languages, that logic belongs under that directory.

Anyways, I am looking forward to it landing in any way you want to do it.  It will help us get to 90% coverage very quickly, which is a magical line in the sand for me.

Reply:
> I dont get why it needs any rewriters. I would put it in visit_Compare

For me it's the complexity. If we look at _visit_AssignOne() in pyrs - it's 110 lines. When we support all the corner cases and are transpiling many real world programs, I worry about how many special cases we have inlined.

Keeping them all in rewriters keeps the base case really simple. For example Assign could just be these lines:

https://github.com/adsharma/py2many/blob/main/pyrs/transpiler.py#L854-L862

everything else gets moved to many small functions.

Anyways, let me fix up the tests and get this in and then we can discuss how to make the code base more maintainable separately.